classes/class-p4-search.php

```diff
@@ -244,12 +304,110 @@ protected function check_cache( $cache_key, $cache_group ) {
 			// If cache key expired then retrieve results once again and re-cache them.
 			if ( false === $this->posts ) {
 				$this->posts = $this->get_timber_posts();
+
 				if ( $this->posts ) {
 					wp_cache_add( $cache_key, $this->posts, $cache_group, self::DEFAULT_CACHE_TTL );
 				}
 			}
 		}
 
+		/**
+		 * Fetches archived results from ArchiveIt.
+		 *
+		 * @param $search_query string search query entered by user.
+		 *
+		 * @return array of archived results.
+		 */
+		protected function get_archived_results( $search_query ) {
+
+			$group = 'archive-search';
+			$subgroup = 'all';
+			$cache_key_set = $this->prepare_keys_for_cache( urldecode( $search_query.'-archived' ), $group, $subgroup );
+			$archive_results = wp_cache_get( $cache_key_set->key, $cache_key_set->group );
+
+			if ( !$archive_results ) {
+				/**
+				 * q = search query.
+				 * s = site.
```

Comment:
Is `s` param is different for all p3 archive results, if yes then could we need a setting for same in P4 setting ?